Development/nmos/client_utils.cpp

```diff
@@ -55,39 +60,165 @@ namespace nmos
                 }
             };
         }
+#endif
+        // get the associated network interface name from the IP address
+        inline utility::string_t get_interface_name(const utility::string_t& address)
+        {
+            utility::string_t interface_name;
+            if (!address.empty())
+            {
+                const auto interfaces = web::hosts::experimental::host_interfaces();
+
+                auto find_interface = [&]()
+                {
+                    return boost::range::find_if(interfaces, [&](const web::hosts::experimental::host_interface& interface)
+                    {
+                        return interface.addresses.end() != boost::range::find(interface.addresses, address);
+                    });
+                };
+                const auto interface = find_interface();
+                if (interfaces.end() != interface) { interface_name = interface->name; }
+            }
+            return interface_name;
+        }
+
+#if !defined(_WIN32) || !defined(__cplusplus_winrt) || defined(CPPREST_FORCE_HTTP_CLIENT_ASIO)
+        // bind socket to a specific network interface, only supporting Linux
+#if defined(linux) || defined(__linux) || defined(__linux__)
```

Comment:
Unnecessary, I believe. We use just `__linux__` in e.g. _cpprest/host_utils.cpp_ and a quick Google found this advice:
http://www.faqs.org/docs/Linux-HOWTO/GCC-HOWTO.html#INDEX.25